### PR TITLE
Rename macro_utils common macros to avoid collision with other libs

### PR DESCRIPTION
**Pull request by ewertons**

### CODE REVIEW

File: macro_utils_h_generator/macro_utils.tt [hunk 1]
```diff
@@ -35,55 +35,55 @@
...
+#define MU_IFCOMMA_0
+#define MU_IFCOMMA_2
 <#for(var i=4;i<=nMacroParameters;i+=2)#>
 <#{#>
-#define IFCOMMA_<#=i#> ,
+#define MU_IFCOMMA_<#=i#> ,
 <#}#>
 
-#define IFCOMMALOGIC(N) C2(IFCOMMALOGIC_, N)
+#define IFCOMMALOGIC(N) MU_C2(IFCOMMALOGIC_, N)
```
Comment (anporumb) [inline macro_utils_h_generator/macro_utils.tt]: >IFCOMMALOGIC [](start = 8, length = 12)

It looks like the intended prefix for the renamed MACROs is "MU_". This macro does not seem to have the "MU_"

File: macro_utils_h_generator/macro_utils.tt [hunk 2]
```diff
@@ -35,55 +35,55 @@
...
 #define IFCOMMALOGIC_0
```
Comment (anporumb) [inline macro_utils_h_generator/macro_utils.tt]: missing "MU_"

File: macro_utils_h_generator/macro_utils.tt [hunk 3]
```diff
@@ -35,55 +35,55 @@
...
-#define IFCOMMA_<#=i#> ,
+#define MU_IFCOMMA_<#=i#> ,
 <#}#>
 
-#define IFCOMMALOGIC(N) C2(IFCOMMALOGIC_, N)
+#define IFCOMMALOGIC(N) MU_C2(IFCOMMALOGIC_, N)
 #define IFCOMMALOGIC_0 
 <#for(var i=1;i<=nMacroParameters;i+=1)#>
 <#{#>
 #define IFCOMMALOGIC_<#=i#> ,
```
Comment (anporumb) [inline macro_utils_h_generator/macro_utils.tt]: >IFCOMMALOGIC_ [](start = 8, length = 13)

missing "MU_"

File: macro_utils_h_generator/macro_utils.tt [hunk 4]
```diff
@@ -35,55 +35,55 @@
...
 <#/*the below produces MOD2 for numbers*/#>
-#define MOD2(x) C2(MOD2_,x)
+#define MU_MOD2(x) MU_C2(MU_MOD2_,x)
 <#for(var i=nArithmetic;i>=0;i--)#>
 <#{#>
-#define MOD2_<#=i#> <#=i%2#>
+#define MU_MOD2_<#=i#> <#=i%2#>
 <#}#>
 
 #define THE_NTH_ARG(<#for(var i=1;i<=nMacroParameters;i++)#>
```
Comment (anporumb) [inline macro_utils_h_generator/macro_utils.tt]: missing "MU_"

**Comment by anporumb** [comment]
> #define _TRIGGER_PARENTHESIS_(...) ,

missing "MU_"

---
Refers to: macro_utils_h_generator/macro_utils.tt:92 in 25de42c. [](commit_id = 25de42cad77eb39cfd84a48db3c826006f5860c5, deletion_comment = False)

**Comment by anporumb** [comment]
> #define LPAREN (

missing "MU_"

---
Refers to: macro_utils_h_generator/macro_utils.tt:94 in 25de42c. [](commit_id = 25de42cad77eb39cfd84a48db3c826006f5860c5, deletion_comment = False)

**Comment by anporumb** [comment]
> #define LPAREN (

Actually I am going to make a "here and everywhere" comment. 

---
In reply to: [467685608](https://github.com/Azure/azure-macro-utils-c/pull/5#issuecomment-467685608) [](ancestors = 467685608)

---
Refers to: macro_utils_h_generator/macro_utils.tt:94 in 25de42c. [](commit_id = 25de42cad77eb39cfd84a48db3c826006f5860c5, deletion_comment = False)

File: macro_utils_h_generator/macro_utils.tt (hunk 1 above)
Comment (ewertons) [reply macro_utils_h_generator/macro_utils.tt]: Addressed.

File: macro_utils_h_generator/macro_utils.tt (hunk 2 above)
Comment (ewertons) [reply macro_utils_h_generator/macro_utils.tt]: Done.

File: macro_utils_h_generator/macro_utils.tt (hunk 3 above)
Comment (ewertons) [reply macro_utils_h_generator/macro_utils.tt]: Addressed.

File: macro_utils_h_generator/macro_utils.tt [hunk 5]
```diff
@@ -318,114 +318,114 @@ IF(1, "true", "false") => "true"
 IF(X, "true", "false") => "true"
```
Comment (anporumb) [inline macro_utils_h_generator/macro_utils.tt]: "MU_"....

File: macro_utils_h_generator/macro_utils.tt [hunk 6]
```diff
@@ -26,290 +26,290 @@
 #include <string.h>
 
 #if (defined OPTIMIZE_RETURN_CODES)
-    #define __FAILURE__ 1
+    #define __MU_FAILURE__ 1
```
Comment (anporumb) [inline macro_utils_h_generator/macro_utils.tt]: >__MU_FAILURE__ [](start = 12, length = 14)

While we are here and a rename will have to follow the complete code base, can you please change this to NOT start with 2 underlines? This is what the C standard has to say about identifiers that start with a double score: Chapter 7.1.3 Reserved identifiers "All identifiers that begin with an underscore and either an uppercase letter or another underscore are always reserved for any use." 

That means we cannot have a __MU_FAILURE__ - you can baptize it to whatever else that is not reserved.

File: macro_utils_h_generator/macro_utils.tt [hunk 7]
```diff
@@ -26,290 +26,290 @@
 #include <string.h>
```
Comment (anporumb) [inline macro_utils_h_generator/macro_utils.tt]: >string [](start = 10, length = 6)

can you please add a #include <stddef.h> here too? As it is needed for the use of size_t

File: macro_utils_h_generator/macro_utils.tt (hunk 6 above)
Comment (YoDaMa) [reply macro_utils_h_generator/macro_utils.tt]: +1 on this

File: macro_utils_h_generator/macro_utils.tt (hunk 4 above)
Comment (ewertons) [reply macro_utils_h_generator/macro_utils.tt]: Done.

File: macro_utils_h_generator/macro_utils.tt (hunk 5 above)
Comment (ewertons) [reply macro_utils_h_generator/macro_utils.tt]: Done. Everywhere.

File: macro_utils_h_generator/macro_utils.tt (hunk 6 above)
Comment (ewertons) [reply macro_utils_h_generator/macro_utils.tt]: Very fair and right. It shall be renamed to MU_FAILURE

File: macro_utils_h_generator/macro_utils.tt (hunk 7 above)
Comment (ewertons) [reply macro_utils_h_generator/macro_utils.tt]: Sure. Done.

**Pull request merged**